diffsims/utils/shape_factor_models.py

```diff
@@ -38,6 +38,7 @@ def binary(excitation_error, max_excitation_error):
     return 1
 
 
+@np.vectorize
```

Comment:
what does this do?

Reply:
makes a scalar function applicable onto an array on an element-wise basis. I found this the easiest way to implement this top-hat function, where if the result is less than 0 it is set to 0. Previously the linear shape factor could be negative, which in my view was actually a bug.

Reply:
Does that mean we get vectorized calculation over `excitation_errors` in `.calculate_ed_data`?

Reply:
yes exactly, you can pass in an array of excitation errors into this function. if `np.vectorized` was not there, it would get hung up on the inequality in the function to set negatives to 0.